### PR TITLE
[mds-stream] Switch to nats.js from ts-nats due to package deprecation

### PR DESCRIPTION
**Pull request by avatarneil**
## 📚 Purpose
We used `ts-nats` over `nats.js` historically due to a more updated API, and better type definitions. Recently, however, `ts-nats` has been deprecated in favor of `nats.js` v2.0, which is now the latest and greatest of TS/JS nats. Unfortunately there were some breaking changes with their API, and a few things had to be updated. The most notable change is that now any messages must be encoded/decoded from UInt8Arrays. I added a wrapper around the processor which is passed in, which decodes NATS messages prior to running the processor function. 

## 👌 Resolves:
- [x] Removes deprecated package

## 📦 Impacts:
- mds-stream

### CODE REVIEW

**Comment by drtyh2o** [comment]
Is there a single place that the consumer transformation can be hooked in?

**Comment by avatarneil** [comment]
@drtyh2o I think so... should be able to change the signature of the `processor` callback that the NatsStreamConsumer takes, and just do the transformations by wrapping the processor.

File: packages/mds-stream/nats/helpers.ts [hunk 1]
```diff
@@ -14,15 +14,15 @@
...
  */
 
-import { connect, MsgCallback, SubscriptionOptions, Client } from 'ts-nats'
+import { connect, SubOpts, Msg, SubscriptionOptions, NatsConnection } from 'nats'
 import logger from '@mds-core/mds-logger'
 import { getEnvVar, asArray } from '@mds-core/mds-utils'
 import { SingleOrArray } from '@mds-core/mds-types'
 
 const initializeNatsClient = () => {
   const { NATS } = getEnvVar({ NATS: 'localhost' })
```
Comment (drtyh2o) [inline packages/mds-stream/nats/helpers.ts:23]: I would love to see this renamed to `NATS_HOST` or `NATS_HOST/NATS_PORT` to be consistent with other services such as `KAFKA_`, `REDIS_`, `PG_`.

File: packages/mds-stream/nats/stream-consumer.ts [hunk 2]
```diff
@@ -14,17 +14,17 @@
...
-import { SubscriptionOptions, Client, MsgCallback } from 'ts-nats'
+import { SubscriptionOptions, NatsConnection, SubOpts, Msg } from 'nats'
 import { Nullable, SingleOrArray } from '@mds-core/mds-types'
 import { createStreamConsumer, disconnectClient } from './helpers'
 import { StreamConsumer } from '../stream-interface'
 
 export const NatsStreamConsumer = (
   topics: SingleOrArray<string>,
-  eachMessage: MsgCallback,
+  eachMessage: SubOpts<Msg>['callback'],
```
Comment (drtyh2o) [inline packages/mds-stream/nats/stream-consumer.ts:24]: I really think we need to handle the `Uint8Array` decoding here.

File: packages/mds-stream/nats/helpers.ts (hunk 1 above)
Comment (avatarneil) [reply packages/mds-stream/nats/helpers.ts:23]: So would I, but I'd rather that be in a targeted PR just for that change ^_^

**Review by drtyh2o — APPROVED** [review]

**Review by nessur — APPROVED** [review]

**Pull request merged**